### PR TITLE
Update Apache 2.0 license and add copyright

### PR DESCRIPTION
**Pull request by raoufchebri**

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -0,0 +1,8 @@
+pg_embedding
+Copyright 2023 Neon Inc.
+
+This product includes software developed at
+Neon Inc. (https://www.neon.tech/).
+
+This product includes software from the ivf-hnsw project (https://github.com/dbaranchuk/ivf-hnsw),
+which is licensed under the MIT License. See the LICENSES/LICENSE.ivf-hnsw file for details.
```
Comment (bayandin) [inline NOTICE:8]: > See the LICENSES/LICENSE.ivf-hnsw file for details.

There's no such file in the PR

Comment (raoufchebri) [reply NOTICE]: Corrected

File: hnswalg.cpp [hunk 2]
```diff
@@ -1,3 +1,19 @@
+/*
+ * Copyright 2023 Neon Inc.
```
Comment (bayandin) [inline hnswalg.cpp:2]: Does this copyright correct for hnswalg.{cpp,h} files?

**Comment by bayandin** [comment]
Closed by https://github.com/neondatabase/pg_embedding/pull/28

**Pull request closed without merging**